### PR TITLE
Add `blended-diffusion`

### PR DESCRIPTION
**Pull request by ydshieh**
[Close for now in order not to run the tests - Re-Open once a more complete version is ready]

Add `blended-diffusion`.

Still WIP, only added some logic regarding `time_embedding_norm` to `ResnetBlock2D` so far.

### CODE REVIEW

**Comment by ydshieh** [comment]
FMI: Regarding `time_embedding_norm`, I looked tha

https://github.com/openai/guided-diffusion/blob/22e0df8183507e13a7813f8d38d51b072ca1e67c/guided_diffusion/unet.py#L248

and https://github.com/huggingface/diffusers/blob/051b34635fda2fc310898a6a602c89be8663b77f/src/diffusers/models/resnet.py#L650-L655

**Comment by ydshieh** [comment]
Hi @patrickvonplaten @patil-suraj @anton-l  I need your opinion regarding the new block type for guided diffusion.

### TL;DR
**Question: I need to add some new arguments to `get_down_block`, but they are only meaningful to guided diffusion. I am not sure if this is the way to go, especially if there are more variations in the future.**



### Details
In [guided-diffusion](https://github.com/openai/guided-diffusion/blob/22e0df8183507e13a7813f8d38d51b072ca1e67c/guided_diffusion/unet.py#L501-L510), some blocks will include attention (but not all)
```
  if ds in attention_resolutions:
      layers.append(
          AttentionBlock(
              ch,
              use_checkpoint=use_checkpoint,
              num_heads=num_heads,
              num_head_channels=num_head_channels,
              use_new_attention_order=use_new_attention_order,
          )
      )
```
With blended-diffusion checkpoint, its UNet model has 6 down blocks, and **the last 3 blocks** will contain attention blocks (along with Res blocks). 

In order to achieve the same, I have to change 
https://github.com/huggingface/diffusers/blob/b56f102765707ce34c7016409a8b901942edb977/src/diffusers/models/unet_blocks.py#L24
to include a new argument `add_attention_block` (similar to `add_downsample`), and calling it in 
https://github.com/huggingface/diffusers/blob/b56f102765707ce34c7016409a8b901942edb977/src/diffusers/models/unet_2d.py#L106-L109
with a condition like `if ds in attention_resolutions:`.

**Comment by ydshieh** [comment]
Well, I am going to move on with the necessary changes for now, and let's decide what to do in the review.

File: src/diffusers/models/resnet.py [hunk 1]
```diff
@@ -289,7 +289,12 @@ def __init__(
         self.conv1 = torch.nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=1, padding=1)
 
         if temb_channels is not None:
-            self.time_emb_proj = torch.nn.Linear(temb_channels, out_channels)
+            if time_embedding_norm == "default":
```
Comment (patrickvonplaten) [inline src/diffusers/models/resnet.py:292]: changes are ok for me!

**Comment by patrickvonplaten** [comment]
Hey @ydshieh,

Sorry to reply only now. The changes look good to me so far. 
I don't think we should add any:

```
if ds in attention_resolutions
```
logic.

Instead we can just define a new `unet_block` in https://github.com/huggingface/diffusers/blob/b56f102765707ce34c7016409a8b901942edb977/src/diffusers/models/unet_blocks.py#L24 no?

**Comment by patrickvonplaten** [comment]
Sorry to leave this without a reply for so long - maybe for now you could just do what you think is right and then we re-iterate in the PR? Will be faster at reviewing from now on :-)

**Comment by ydshieh** [comment]
> if ds in attention_resolutions

No problem. I will do whatever I need, and let's discuss what to change once a complete draft PR is ready.
But just FYI:

> define a new unet_block in get_down_block( 
Idea is so, but due to the difference between the new block and the existing ones, `get_down_block` needs more arguments, and some of them only make senses for the new model. But let's discuss it later.

**Comment by patrickvonplaten** [comment]
It's fine to add new arguments - there shouldn't be a problem with that :-)

**Comment by ydshieh** [comment]
**[Close for now in order not to run the tests - Re-Open once a more complete version is ready]**
